### ISSUE TITLE
cortexm_common: use of r13 is deprecated

### ISSUE THREAD

**Issue by authmillenon** [RIOT-OS/RIOT#3664]
Since the merge of #3333 I get the following warning:

```
$ BOARD=iot-lab_M3 IOTLAB_NODES=2 make -C examples/gnrc_networking/ all
make: Entering directory `/home/martine/Repositories/RIOT-OS/RIOT/examples/gnrc_networking'
Building application "gnrc_networking" for "iot-lab_M3" with MCU "stm32f1".

"make" -C /home/martine/Repositories/RIOT-OS/RIOT/boards/iot-lab_M3
"make" -C /home/martine/Repositories/RIOT-OS/RIOT/core
"make" -C /home/martine/Repositories/RIOT-OS/RIOT/cpu/stm32f1
"make" -C /home/martine/Repositories/RIOT-OS/RIOT/cpu/cortexm_common
/tmp/ccwJhJJ3.s: Assembler messages:
/tmp/ccwJhJJ3.s:211: Warning: use of r13 is deprecated
"make" -C /home/martine/Repositories/RIOT-OS/RIOT/cpu/stm32f1/periph
[…]
```

**Comment by gebart** [comment]
I will look into this.

**Comment by thomaseichinger** [comment]
The line causing this is [vectors_cortexm.c:137](https://github.com/RIOT-OS/RIOT/blob/master/cpu/cortexm_common/vectors_cortexm.c#L137).

`cmp r0, sp` doesn't seem suspicious on first sight.

**Comment by OlegHahm** [comment]
I think `make scan-build` was reporting this since it was introduced.

**Comment by thomaseichinger** [comment]
Oops, `r13` is `sp`. Forgot about this. :grimacing:

**Comment by daniel-k** [comment]
I found [this](https://sourceware.org/ml/binutils/2009-02/msg00298.html) after googling. But I couldn't find the ARM docs yet, that mention this.

**Comment by thomaseichinger** [comment]
Found this too, `cmp sp, r0` would be allowed.

**Comment by gebart** [comment]
yes, the solution is to mov r1, sp first, then cmp r0, r1

**Comment by thomaseichinger** [comment]
Some register juggling. :smile:

**Comment by thomaseichinger** [comment]
As [reference](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0489g/Cihiddid.html)
> **Use of SP in ARM and Thumb instructions**
> You can use SP for Rn in ARM and Thumb instructions.
> You can use SP for Rm in ARM instructions but this is deprecated in ARMv6T2 and above.
> You can use SP for Rm in a 16-bit Thumb CMP Rn, Rm instruction but this is deprecated in ARMv6T2 and above. Other use of SP for Rm is not permitted in Thumb.

**Comment by authmillenon** [comment]
@PeterKietzmann please explain why you closed this issue.

**Comment by PeterKietzmann** [comment]
It was automatically closed after I merged #3666. Isn't it fixed for you?

**Comment by authmillenon** [comment]
Ah, sorry. Somehow I landed in #3356 and this wasn't merged yet.
